airflow/providers/snowflake/operators/snowflake.py

```diff
@@ -125,3 +126,285 @@ def execute(self, context: Any) -> None:
 
         if self.do_xcom_push:
             return execution_info
+
+
+class _SnowflakeDbHookMixin:
```

Comment:
Why a class and not just a method inside the SnowflakeCheckOperator?

Reply:
Good point

Reply:
Trying to be more DRY. Originally I tried making the `Check` classes inherit from just the `SnowflakeOperator`, but it resulted in an error in the constructors.

Reply:
DRY philosophy is great. However, my point was that you don't need a class at all and you can get rid of the class extra layer of abstraction. My comment is inspired by https://www.youtube.com/watch?v=o9pEzgHorH0

Using the same class too in the SnowflakeOperator has been quite an improvement, but I think you can still have just a function in the file, and you can just call it from the operators. Anyway, it's not a big deal and the PR is already approved, so up to u.

Reply:
I wached the presentation :). Entertaining and I mostly agree with it  (not everything :) 

But yeah - in  this case using mixin is definitely over-the-top.

Reply:
Would you mind changing that @denimalpaca ?

Reply:
Will have this fixed Tuesday, looks like with the way BaseSQLOperator works, even the function without a class might not be necessary.